druntime/src/core/stdc/stdatomic.d

```diff
@@ -1116,9 +1192,48 @@ A atomic_fetch_and_explicit_impl(A, M)(shared(A)* obj, M arg, memory_order order
 }
 
 ///
-unittest
+@trusted unittest
 {
     shared(int) val = 5;
     atomic_fetch_and_explicit_impl(&val, 3, memory_order.memory_order_seq_cst);
     assert(atomic_load_impl(&val) == 1);
 }
+
+private:
+
+pragma(inline, true)
+A atomic_fetch_op(memory_order order, string op, A, M)(A* obj, M arg) @trusted
+{
+    static if (is(A : ulong) && (op == "+=" || op == "-="))
+    {
+        // these cannot handle floats
+        static if (op == "+=")
+        {
+            return atomicFetchAdd!order(obj, arg);
+        }
+        else static if (op == "-=")
+        {
+            return atomicFetchSub!order(obj, arg);
+        }
+    }
+    else
+    {
+        // copied from core.atomic
+        static if (op == "+=" || op == "-="  || op == "*="  || op == "/=" ||
+                op == "%=" || op == "^^=" || op == "&="  || op == "|=" ||
+                op == "^=" || op == "<<=" || op == ">>=" || op == ">>>=") // skip "~="
+        {
```

Comment:
This list seems unnecessary when you just need `+=` or `-=`

Reply:
Currently yes.

I need to cleanup other parts that do this operation manually.

Reply:
Some of those are now used.